### PR TITLE
Task/generation outside

### PR DESCRIPTION
**Pull request by M1croZavr**

### CODE REVIEW

File: src/compression_horizon/metric.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            1,
+            generated_token_log_probs.argmax(dim=1).view(-1, 1),
+        ).mean()
+    )
+    perplexity = torch.exp(cross_entropy).item()
+    return perplexity
+
+
+@torch.no_grad()
+def calculate_perplexity_logits(
```
[the remark this answers is not in the document]
Comment (M1croZavr) [reply src/compression_horizon/metric.py]: Используется как метрика confidence score в notebooks/Compression_generation_outside.ipynb
